hubblestack/files/hubblestack_nova/oval_scanner.py

```diff
@@ -108,44 +122,38 @@ def parse_impact_report(report, local_pkgs, hubble_format, impacted_pkgs=[]):
 
 def write_report_to_file(opt_output_file, report):
     """Write report to local disk"""
+    time = datetime.datetime.now()
+    now = time.strftime("%Y-%m-%d@%H:%M:%S")
     logging.info('Writing CVE data to {0}'.format(opt_output_file))
+    if __salt__['file.file_exists'](opt_output_file):
+        logging.info('Found existing log, backing up...')
+        back_up = opt_output_file + now
+        __salt__['file.rename'](opt_output_file, back_up)
     with open(opt_output_file, 'w') as outfile:
         outfile.write(json.dumps(report, indent=2, sort_keys=True))
 
 
-def get_impact_report(vulns, local_pkgs, distro_name):
+def get_impact_report(vulns, local_pkgs, distro_name, queue):
     """Get impact report"""
     logging.debug('get_impact_report')
-    report = build_impact(vulns, local_pkgs, distro_name)
+    report = build_impact(vulns, local_pkgs, distro_name, queue)
     logging.debug(json.dumps(report, indent=4, sort_keys=True))
     return report
 
 
 # Build an impact report
-def build_impact(vulns, local_pkgs, distro_name, result={}):
-    """Build impacts based on pkg comparisons"""
+def build_impact(vulns, local_pkgs, distro_name, queue, result={}):
+    """Build impact"""
     logging.debug('build_impact')
-    for data in vulns.values():
-        for pkg in data['pkg']:
-            name = pkg['name']
-            ver = pkg['version']
-            if name in local_pkgs:
-                title = data['title']
-                cve = data['cve']
-                if 'severity' in data:
-                  severity = data['severity']
-                else:
-                  severity = 'N/A'
-                if distro_name in ('centos', 'redhat'):
-                    advisory = data['rhsa']
-                else:
-                    if 'advisories' in data:
-                      advisory = data['advisories']
-                    else:
-                      advisory = cve
-                impact = get_impact(local_pkgs[name], name, ver, title, cve, advisory, severity)
-                if impact:
-                    result = build_impact_report(impact)
+    determine_impact(vulns, local_pkgs, distro_name, queue)
+    while not queue.empty():
+        impact = queue.get()
+        if impact:
+            if 'kernel-aarch64' in list(impact.keys())[0]:
```

Comment:
You should put some comments about the purpose of this condition. As a reviewer, I don't get any idea why this condition is here.
Probably if you read this code after 6 months, you will also wonder why this condition exists?